File: streaming-service/models/song.go
package models

import "time"

type Song struct {
	ID          uint `gorm:"primaryKey"`
	AlbumID     uint `gorm:"not null;index"`
	GenreID     uint `gorm:"not null;index"`
	Title       string
	Duration    int // segundos
	AudioURL    string
	TrackNumber int
	CreatedAt   time.Time
	UpdatedAt   time.Time

	Album   Album
	Genre   Genre
	Artists []Artist `gorm:"many2many:song_artists"`
}

File: content-service/core/entities/song.py
from pydantic import BaseModel
from datetime import date
from typing import Optional, List

from .artist import ArtistOut  # forward refs


class SongBase(BaseModel):
    title: str
    duration: int
    audio_url: str
    track_number: Optional[int] = None


class SongCreate(SongBase):
    album_id: int
    genre_id: Optional[int] = None
    artist_ids: List[int]  # para la relación many-to-many


class SongUpdate(SongBase):
    album_id: Optional[int] = None
    genre_id: Optional[int] = None


class SongOut(SongBase):
    id: int
    album_id: int
    genre_id: Optional[int]
    created_at: Optional[date]
    updated_at: Optional[date]
    # artists: List["ArtistOut"] = []

    class Config:
        from_attributes = True
